src/main/java/io/appium/java_client/remote/AppiumCommandExecutor.java

```diff
@@ -147,6 +186,12 @@ protected HttpClient getClient() {
         return getPrivateFieldValue(HttpCommandExecutor.class, "client", HttpClient.class);
     }
 
+    protected void setClientWithURL(URL serverUrl) {
+        HttpClient.Factory httpClientFactory = getPrivateFieldValue(HttpCommandExecutor.class,
+                "httpClientFactory", HttpClient.Factory.class);
+        setPrivateFieldValue(HttpCommandExecutor.class, "client", httpClientFactory.createClient(serverUrl));
```

Comment:
I'm not quite sure we should create a new client here. Can we simply replace the url for the existing one?

Reply:
I don't see any difference as we anyway must access a private field here